src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,369 @@
+             const DataModel::Nullable<FloorSurfaceTag> & aSurfaceTag)
+    {
+        locationID = aLocationId;
+        mapID      = aMapId;
+
+        // If there is at least one non-null value for locationInfo, add it to the location structure.
+        if ((!aLocationName.empty()) || (!aFloorNumber.IsNull()) || (!aAreaType.IsNull()))
+        {
+            // Create a home location info structure and fill it in except for the location name. This is done below.
+            locationInfo.locationInfo.SetNonNull(Structs::HomeLocationStruct::Type());
+
+            locationInfo.locationInfo.Value().floorNumber = aFloorNumber;
+            locationInfo.locationInfo.Value().areaType    = aAreaType;
+        }
+        else
+        {
+            locationInfo.locationInfo.SetNull();
+        }
+
+        locationInfo.landmarkTag = aLandmarkTag;
+        locationInfo.positionTag = aPositionTag;
+        locationInfo.surfaceTag  = aSurfaceTag;
+
+        // Save the location name to buffer, so access and lifetime is controlled.
+        // If aLocationName is not null, create a CharSpan to reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
+        memset(mLocationNameBuffer, 0, sizeof(mLocationNameBuffer));
+
+        // this assumes locationInfo structure was created above, if appropriate
+        if (!locationInfo.locationInfo.IsNull())
+        {
+            if (aLocationName.empty())
+            {
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, 0);
+            }
+            else if (aLocationName.size() > sizeof(mLocationNameBuffer))
+            {
+                // Save the truncated name that fits into available size.
+                memcpy(mLocationNameBuffer, aLocationName.data(), sizeof(mLocationNameBuffer));
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, sizeof(mLocationNameBuffer));
+            }
+            else
+            {
+                // Save full name.
+                memcpy(mLocationNameBuffer, aLocationName.data(), aLocationName.size());
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, aLocationName.size());
+            }
+        }
+    }
+
+    /**
+     * @brief Compare the location's name with the given text.
+     * @param[in] aLocationName The name to compare.
+     * @return true if the location structure's name field matches aLocationName.
+     *
+     * @note if locations structure's name field is null, returns false.
+     */
+    bool IsNameEqual(const CharSpan & aLocationName) const
+    {
+        if (!locationInfo.locationInfo.IsNull())
+        {
+            return locationInfo.locationInfo.Value().locationName.data_equal(aLocationName);
+        }
+
+        return false; // name is Null or text sizes do not match
```

Comment:
This one is a bit weird.  Per spec, null and empty string should always be treated identically, no?

Reply:
I think the comment is a bit incorrect here. The string itself cannot be null. It's the structure within which the string exists that can be null. I have amended the doc to make the clear.